dev/cuda/layernorm_backward.cu

```diff
@@ -129,6 +129,13 @@ void layernorm_backward_cpu(float* dinp, float* dweight, float* dbias,
 // GPU kernels
 
 // GPU helper functions for atomicAdd on smaller than 32-bit types
+__device__ floatX warpReduceSum(floatX val) {
```

Comment:
maybe we should move this to common.h; I guess we're going to use it for most of the kernels that currently have cgs

Reply:
maybe even a template version, then we have full feature parity with cg::reduce

Reply:
I agree we should move this to common.h